src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs

```diff
@@ -2931,7 +2931,48 @@ public static void DCJS_ValidateExceptionOnUnspecifiedRootSerializationType()
         
         Assert.Equal(value.MyIntProperty, actual.MyIntProperty);
         Assert.Equal(value.MyStringProperty, actual.MyStringProperty);
-    } 
+    }
+
+    [Fact]
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "On netfx throws System.Runtime.CallbackException")]
+    public static void DSJS_ThrowExceptionOnDispose()
+    {
+        using (MemoryStream ms = new MemoryStream(System.Text.Encoding.Unicode.GetBytes("{}")))
+        {
+            XmlDictionaryReader jsonReader = JsonReaderWriterFactory.CreateJsonReader(ms, System.Text.Encoding.Unicode, XmlDictionaryReaderQuotas.Max,
+                reader =>
+                {
+                    //sample exception on reader close
+                    throw new DivideByZeroException();
+                });
+
+            InvalidOperationException exception = Assert.Throws<InvalidOperationException>(() => jsonReader.Dispose());
+            Assert.IsType(typeof(DivideByZeroException), exception.InnerException);
+        }
+    }
+
+    [Fact]
+    [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp, "On netcoreapp throws InvalidOperationException")]
+    public static void DSJS_ThrowExceptionOnDispose_Netfx()
```

Comment:
This test failed when the test was targeting UWP CoreCLR as the `SkipOnTargetFramework` above prevent the test from running on `netcoreapp` only.

We can unify the tests by checking if `innerException is InvalidOperationException || innerException.GetType().FullName == "System.Runtime.CallbackException"`

Reply:
done